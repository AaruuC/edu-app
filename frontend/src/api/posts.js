import axios from 'axios';

export const getAllClassrooms = async () => {
  try {
    const response = await axios.get('/classroom');
    return response.data;
  } catch (err) {
    return err;
  }
};

export const createNewClassroom = async (classroomObject) => {
  try {
    const {
      name, admins, users, posts,
    } = classroomObject;
    const response = await axios.post('/classroom/create', {
      name,
      admins,
      users,
      posts,
    });
    console.log(response);
    return response;
  } catch (err) {
    return err;
  }
};

export const getAllPostsInClassroomById = async (id) => {
  try {
    const response = await axios.get('/classroom/getId', {
      params: { id },
    });
    console.log(response);
    return response.data;
  } catch (err) {
    return err;
  }
};

// export const getPostById = async (id) => {
//   try {
//     const response = await axios.get(${rootURL}/${id});
//     // console.log('a post', response.data);
//     return response.data;
//   } catch (err) {
//     // console.error('error', err.message);
//     return err;
//   }
// };

export const createNewPost = async (classroomId, postObject) => {
  try {
    const {
      title, body, priv, anonymous, replies,
    } = postObject;
    const response = await axios.post('/classroom/addPost', {
      title, body, priv, id: classroomId, anonymous, replies,
    });
    console.log(response);
    // console.log('a response', response.data);
    return response;
  } catch (err) {
    // console.error('error', err.message);
    return err;
  }
};
