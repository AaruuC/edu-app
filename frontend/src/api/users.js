import axios from 'axios';
// import { userURL } from '../utils/utils';

// export const getAllUsers = async () => {
//   try {
//     const response = await axios.get(userURL);
//     return response.data;
//   } catch (err) {
//     return err;
//   }
// };

// export const getUserById = async (id) => {
//   try {
//     const response = await axios.get(`${userURL}/${id}`);
//     // console.log('a post', response.data);
//     return response.data;
//   } catch (err) {
//     // console.error('error', err.message);
//     return err;
//   }
// };

const createUser = async (userObject) => {
  console.log('atapi');
  try {
    if (userObject.username === '' || userObject.password === '') {
      throw new Error('invalid username or password');
    }
    const response = await axios.post('/account/signup', {
      username: userObject.username,
      password: userObject.password,
    });
    console.log(response);
    return response;
  } catch (err) {
    console.error('error', err.message);
    return err;
  }
};

export default createUser;
