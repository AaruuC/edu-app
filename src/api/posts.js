import axios from 'axios';
import { classroomURL } from '../utils/utils';

export const getAllClassrooms = async () => {
  try {
    const response = await axios.get(classroomURL);
    return response.data;
  } catch (err) {
    return err;
  }
};

export const createNewClassroom = async (classroomObject) => {
  try {
    const {
      name, admins, users, posts,
    } = classroomObject;
    const response = await axios.post(classroomURL, {
      name,
      admins,
      users,
      posts,
    });
    return response.data;
  } catch (err) {
    return err;
  }
};

export const getAllPostsInClassroomById = async (id) => {
  try {
    const response = await axios.get(`${classroomURL}/${id}`);
    return response.data;
  } catch (err) {
    return err;
  }
};

// export const getPostById = async (id) => {
//   try {
//     const response = await axios.get(`${rootURL}/${id}`);
//     // console.log('a post', response.data);
//     return response.data;
//   } catch (err) {
//     // console.error('error', err.message);
//     return err;
//   }
// };

export const createNewPost = async (classroomId, postObject) => {
  try {
    const response = await axios.post(`${classroomURL}/${classroomId}`, {
      title: postObject.title,
      body: postObject.body,
      private: postObject.private,
      anonymous: postObject.anonymous,
    });
    // console.log('a response', response.data);
    return response.data;
  } catch (err) {
    // console.error('error', err.message);
    return err;
  }
};
