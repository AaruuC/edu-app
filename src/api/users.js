import axios from 'axios';
import { userURL } from '../utils/utils';

export const getAllUsers = async () => {
  try {
    const response = await axios.get(userURL);
    return response.data;
  } catch (err) {
    return err;
  }
};

export const getUserById = async (id) => {
  try {
    const response = await axios.get(`${userURL}/${id}`);
    // console.log('a post', response.data);
    return response.data;
  } catch (err) {
    // console.error('error', err.message);
    return err;
  }
};

export const createUser = async (userObject) => {
  try {
    if (userObject.username === '' || userObject.password === '') {
      throw new Error('invalid username or password');
    }
    const response = await axios.post(userURL, {
      username: userObject.username,
      password: userObject.password,
    });
    // useNavigate('/');
    // window.location = '/';
    return response.data;
  } catch (err) {
    console.error('error', err.message);
    return err;
  }
};
