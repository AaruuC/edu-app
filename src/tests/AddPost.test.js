/* eslint-disable no-undef */
// import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
// import renderer, { act } from 'react-test-renderer';
// import { test, expect } from '@testing-library/jest-dom';
// import React from 'react';
// import AddPost from '../components/AddPost';

// const posts = [
//   {
//     id: 1,
//     title: 'hw3 q5 help',
//     body: 'help',
//     private: true,
//     anonymous: false,
//   },
//   {
//     id: 2,
//     title: 'hw1 q7 help',
//     body: 'help',
//     private: true,
//     anonymous: false,
//   },
//   {
//     id: 3,
//     title: 'hw5 q5 help',
//     body: 'help',
//   },
//   {
//     id: 4,
//     title: 'hw3 q1 help',
//     body: 'help',
//   },
//   {
//     title: 'hw1 q1 help',
//     body: 'help',
//     id: 5,
//   },
// ];

// const setPosts = () => {};

// test('renders addPost has private', () => {
//   render(<AddPost />);
//   const linkElement = screen.getByLabelText('Private');

//   expect(linkElement).toBeInTheDocument();
// });

// test('renders addPost has anonymous', () => {
//   render(<AddPost />);
//   const linkElement = screen.getByLabelText('Anonymous');

//   expect(linkElement).toBeInTheDocument();
// });

// test('renders addPost has title', () => {
//   render(<AddPost />);
//   const linkElement = screen.getByLabelText('Title');
//   expect(linkElement).toBeInTheDocument();
// });
// test('renders addPost has id', () => {
//   render(<AddPost />);
//   const linkElement = screen.getByTestId('body');

//   expect(linkElement).toBeInTheDocument();
// });
// test('renders addPost matches snapshot', () => {
//   const component = renderer.create(<AddPost />);
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('renders addPost has form-group', () => {
//   render(<AddPost />);
//   const linkElement = screen.getByTestId('form-group');
//   expect(linkElement).toHaveClass('form-group');
//   expect(linkElement).toBeInTheDocument();
// });

// test('renders addPost has button', () => {
//   render(<AddPost />);
//   const linkElement = screen.getByTestId('button');
//   expect(linkElement).toHaveClass('btn btn-primary');
//   expect(linkElement).toBeInTheDocument();
// });

// test('renders addPost button click', async () => {
//   render(<AddPost posts={posts} addNewPost={setPosts} />);
//   const linkElement1 = screen.getByLabelText('Title');
//   const linkElement2 = screen.getByTestId('body');
//   const linkElement3 = screen.getByRole('button', { name: /Create Post/ });
//   expect(linkElement2).toBeInTheDocument();
//   await act(async () => {
//     await userEvent.type(linkElement1, 'asd');
//     await userEvent.type(linkElement2, 'need help for this question');
//   });
//   expect(screen.getByDisplayValue('asd')).toBeInTheDocument();
//   expect(screen.getByDisplayValue('need help for this question')).toBeInTheDocument();
//   await act(async () => {
//     await userEvent.click(linkElement3);
//   });

//   expect(linkElement1.value).toBe('');
//   expect(linkElement2.value).toBe('');
// });
import axios from 'axios';
import { createNewPost } from '../api/posts';

jest.mock('axios');

test('createNewPost add a new post to the classroom', async () => {
  const id = '5';
  const postObject = {
    title: 'q1',
    body: 'need help',
    private: false,
    id: '6',
    anonymous: true,
    replies: [],
  };
  const classroomData = {
    id,
    posts: [],
  };

  axios.get.mockResolvedValueOnce({ data: classroomData });
  axios.put.mockResolvedValueOnce({ data: { ...classroomData, posts: [postObject] } });

  const response = await createNewPost(id, postObject);

  expect(response).toEqual({ ...classroomData, posts: [postObject] });
  expect(axios.get).toHaveBeenCalledWith(`${classroomURL}/${id}`);
  expect(axios.put).toHaveBeenCalledWith(`${classroomURL}/${id}`, { ...classroomData, posts: [postObject] });
});
