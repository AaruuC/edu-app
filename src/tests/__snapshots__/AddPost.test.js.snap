// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders addPost matches snapshot 1`] = `
<div>
   
  <form
    className="mx-auto"
    id="add"
    style={
      Object {
        "width": "800px",
      }
    }
  >
    <div
      className="form-group"
    >
      <label
        className="form-check-label"
        htmlFor="Title"
      >
        Title
      </label>
      <input
        className="form-control"
        id="Title"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
      data-testid="form-group"
    >
      <div
        className="input-group-prepend"
      />
      <textarea
        aria-label="With textarea"
        className="form-control"
        data-testid="body"
        onChange={[Function]}
      />
    </div>
    <div
      className="form-row"
    >
      <div
        className="form-check col-md-4"
      >
        <input
          className="form-check-input"
          id="Private"
          onClick={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="Private"
        >
          Private
        </label>
      </div>
      <div
        className="form-check col-md-6 "
      >
        <input
          className="form-check-input"
          id="Anonymous"
          onClick={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="Anonymous"
        >
          Anonymous
        </label>
      </div>
      <button
        className="btn btn-primary"
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        Create Post
      </button>
    </div>
  </form>
</div>
`;
